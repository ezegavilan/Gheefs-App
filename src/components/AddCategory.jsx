import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({ target: { value } }) => {
        setInputValue(value);
    }

    const addCategory = (event) => {
        event.preventDefault();

        const value = inputValue.trim();

        if (value <= 1) return;

        // onAddCategory(categories => [...categories, inputValue]);
        onNewCategory(value);
        setInputValue('');
    }

    return (
        <form aria-label="form" onSubmit={ addCategory }>
            <input
                type="text"
                placeholder="Buscar gifs"
                value={ inputValue }
                onChange={ onInputChange }
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}
