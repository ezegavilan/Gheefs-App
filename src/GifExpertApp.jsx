import { useState } from 'react';

import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['The Last of Us', 'God of War']);

    const onAddCategory = (category) => {
        /* Con spread operator
        setCategories([...categories, 'Valorant']);
         */

        /* Utilizando el callback
        setCategories(category => [...category, 'Valorant']);
         */

        if (categories.includes(category)) return;

        setCategories([category, ...categories]);
    }

    return (
        <>
            <h1>Â¡Bienvenido a Gheefs!</h1>

            <AddCategory onNewCategory={ onAddCategory }/>

            {
                categories.map(category => (
                        <GifGrid key={ category } category={ category }/>
                    )
                )
            }

        </>
    );
}
