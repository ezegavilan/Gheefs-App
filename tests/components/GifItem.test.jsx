import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components";


describe('Tests on <GifItem />', () => {
    const title = 'TÃ­tulo';
    const url = 'https://url.example.com/image.png';

    test('Should match with snapshot', () => {
        const { container } = render(<GifItem title={ title } url={ url }/>);

        expect(container).toMatchSnapshot();
    });

    test('Should show image with url and ALT specified', () => {
        render(<GifItem title={ title } url={ url }/>);

        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);

        const { src, alt } = screen.getByRole('img');

        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('Should show title on component', () => {
        render(<GifItem title={ title } url={ url }/>);

        expect(screen.getByText(title)).toBeTruthy();
    });
})
