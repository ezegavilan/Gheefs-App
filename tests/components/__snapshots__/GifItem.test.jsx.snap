// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifItem /> Should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Título"
      src="https://url.example.com/image.png"
    />
    <p>
      Título
    </p>
  </div>
</div>
`;
